app/src/main/java/nerd/tuxmobil/fahrplan/congress/settings/SettingsFragment.kt

```diff
@@ -4,44 +4,64 @@ import android.annotation.TargetApi
-import android.preference.Preference.OnPreferenceClickListener
-import android.preference.PreferenceCategory
-import android.preference.PreferenceFragment
-import android.preference.PreferenceScreen
 import android.provider.Settings
 import androidx.core.os.bundleOf
+import androidx.preference.EditTextPreference
+import androidx.preference.Preference
+import androidx.preference.Preference.OnPreferenceChangeListener
+import androidx.preference.Preference.OnPreferenceClickListener
+import androidx.preference.PreferenceCategory
+import androidx.preference.PreferenceFragmentCompat
+import androidx.preference.PreferenceScreen
+import androidx.preference.SwitchPreferenceCompat
 import nerd.tuxmobil.fahrplan.congress.BuildConfig
 import nerd.tuxmobil.fahrplan.congress.R
 import nerd.tuxmobil.fahrplan.congress.alarms.AlarmServices
 import nerd.tuxmobil.fahrplan.congress.contract.BundleKeys
 import nerd.tuxmobil.fahrplan.congress.extensions.getAlarmManager
 import nerd.tuxmobil.fahrplan.congress.extensions.toSpanned
+import nerd.tuxmobil.fahrplan.congress.preferences.AlarmTonePreference
+import nerd.tuxmobil.fahrplan.congress.repositories.AppRepository
 import nerd.tuxmobil.fahrplan.congress.utils.FahrplanMisc
 
-class SettingsFragment : PreferenceFragment() {
+class SettingsFragment : PreferenceFragmentCompat() {
+
+    private companion object {
+
+        const val REQUEST_CODE_ALARM_TONE = 3439
+
+    }
+
+    private lateinit var appRepository: AppRepository
+
+    override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
+        addPreferencesFromResource(R.xml.prefs)
+    }
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        addPreferencesFromResource(R.xml.prefs)
+        appRepository = AppRepository
 
-        val categoryGeneral = findPreference(getString(R.string.preference_key_category_general)) as PreferenceCategory
+        val categoryGeneral = requirePreference<PreferenceCategory>(getString(R.string.preference_key_category_general))
 
-        findPreference(resources.getString(R.string.preference_key_auto_update_enabled)).onPreferenceChangeListener = OnPreferenceChangeListener { _: Preference?, newValue: Any ->
+        requirePreference<SwitchPreferenceCompat>(resources.getString(R.string.preference_key_auto_update_enabled)).onPreferenceChangeListener = OnPreferenceChangeListener { _: Preference?, newValue: Any ->
             val isAutoUpdateEnabled = newValue as Boolean
             if (isAutoUpdateEnabled) {
                 FahrplanMisc.setUpdateAlarm(activity, true)
             } else {
-                val alarmManager = activity.getAlarmManager()
-                AlarmServices(alarmManager).discardAutoUpdateAlarm(activity)
+                with(requireActivity()) {
+                    val alarmManager = getAlarmManager()
+                    AlarmServices(alarmManager).discardAutoUpdateAlarm(this)
+                }
```

Comment:
Have you considered using a dependency injection framework to clean up code like this?

Reply:
Up to now I don't have the feeling that a DI framework is needed.